### PR TITLE
Implement `Math.lcm` TypeScript goody

### LINKED ISSUE #86: Add LCM to the adventure pack
It would be nice to have a least common multiple function available in the adventure pack. This can be implemented based on the GCD (see also #85) and we can have it at `Math.lcm` as a starting point.

### PR DESCRIPTION
**Pull request by simona1**
Closes #86.

### CODE REVIEW

**Comment by simona1** [comment]
Lol, is it goody or goodie?

File: workspaces/adventure-pack/goodies/typescript/Math.lcm/index.test.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    [-5, 15, 15],
+    [7, -3, 21],
+    [-6, -8, 24],
+    [1000000007n, 1000000009n, 1000000016000000063n],
+    [2n, 1000000007n, 2000000014n],
+    [9999991n, 99999959n, 999998690000369n],
+    [123456, 789012, 8117355456],
+    [15, 25, 75],
+  ])("Math.lcm(%p, %p) === %p", (a, b, expected) => {
+    expect(Math.lcm(Number(a), Number(b))).toBe(Number(expected));
```
Comment (miorel) [inline workspaces/adventure-pack/goodies/typescript/Math.lcm/index.test.ts:23]: Can we avoid the Number casts, and express the numbers above not in `bigint` form? If the numbers are too big to express as regular Numbers, the result is likely not meaningful anyway.

File: workspaces/adventure-pack/goodies/typescript/Math.lcm/index.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+}
+
+Math.lcm = function (a: number, b: number): number {
+  if (a === 0 || b === 0) {
+    return 0;
+  }
+  a = Math.abs(a);
+  b = Math.abs(b);
+
+  return (a / Math.gcd(a, b)) * b;
```
Comment (miorel) [inline workspaces/adventure-pack/goodies/typescript/Math.lcm/index.ts:16]: This whitespace style feels slightly more natural?

```suggestion
  }

  a = Math.abs(a);
  b = Math.abs(b);
  return (a / Math.gcd(a, b)) * b;
```

**Review by miorel — APPROVED** [review]

File: workspaces/adventure-pack/goodies/typescript/Math.lcm/index.test.ts (hunk 1 above)
Comment (simona1) [reply workspaces/adventure-pack/goodies/typescript/Math.lcm/index.test.ts:23]: ok, I will remove the offending test case (line 17 actually gave me a warning like this: 
```
 This number literal will lose precision at runtime  no-loss-of-precision
```

File: workspaces/adventure-pack/goodies/typescript/Math.lcm/index.ts [hunk 3]
```diff
@@ -0,0 +1,21 @@
+import "../Math.gcd";
+
+declare global {
+  interface Math {
+    lcm(a: number, b: number): number;
+  }
+}
+
+Math.lcm = function (a: number, b: number): number {
+  if (a === 0 || b === 0) {
```
Comment (miorel) [inline workspaces/adventure-pack/goodies/typescript/Math.lcm/index.ts:10]: As a fun aside, this is a reason why arguably gcd(0, 0) should be infinite. Then when we delete by Infinity below, we naturally get 0 for the lcm 😄

**Pull request merged**